core/domain/src/main/java/com/pronoidsoftware/core/domain/agendaitem/AgendaItem.kt

```diff
@@ -0,0 +1,41 @@
+package com.pronoidsoftware.core.domain.agendaitem
+
+import kotlinx.datetime.LocalDateTime
+
+sealed class AgendaItem(
+    // null if new
+    open val id: String?,
```

Comment:
I'm thinking if this is the right way to implement whether an item is new or not. Having such a null field in a domain model would also tell that the ID is just an optional field for agenda items (which it isn't really - every valid agenda item that can be displayed must have an ID).

Since we need to generate these IDs client-side anyways, could it make sense to make this field non-null and rather work with a separate boolean `isNew` or so which you pass to your create event call?

Reply:
yeah, I didn't like having it null all that much either. However, I don't think I need to persist the isNew Boolean within the domain model itself. I've created https://github.com/evanadwyer-pronoid-software/Tasky/pull/96 with a mechanism that I think works for me